lib/jwt/decode.rb

```diff
@@ -32,12 +34,15 @@ def decode_segments
 
     private
 
+    def verify_algo
+      raise(JWT::IncorrectAlgorithm, 'An algorithm must be specified') if allowed_algorithms.empty?
+      raise(JWT::IncorrectAlgorithm, 'Expected a different algorithm') unless options_includes_algo_in_header?
+    end
+
     def verify_signature
       @key = find_key(&@keyfinder) if @keyfinder
       @key = ::JWT::JWK::KeyFinder.new(jwks: @options[:jwks]).key_for(header['kid']) if @options[:jwks]
-
-      raise(JWT::IncorrectAlgorithm, 'An algorithm must be specified') if allowed_algorithms.empty?
-      raise(JWT::IncorrectAlgorithm, 'Expected a different algorithm') unless options_includes_algo_in_header?
+      @key = JWT::X5cKeyFinder.from(header['x5c'], @options.fetch(:root_certificates), @options[:crls]) if header['x5c']
```

Comment:
Think the trigger for this needs to be indicated by the application requesting the verification. Now it seems to be only the `x5c` header that controls when this happens.

Also a test for this path would be valuable.

Maybe something like this could trigger the x5c logic.

```
JWT.decode(token, nil, true, x5c: {root_certificates: [], crls: [] })
```